### PR TITLE
docs: document release QA procedure

### PR DESCRIPTION
**Pull request by rudolfs**
This formalises our successful trial QA procedure run https://github.com/radicle-dev/radicle-upstream/issues/912.

Closes: https://github.com/radicle-dev/radicle-upstream/issues/842.
Closes: https://github.com/radicle-dev/radicle-upstream/issues/910.

### CODE REVIEW

File: DEVELOPMENT.md [hunk 1]
```diff
@@ -482,6 +490,30 @@ Finalizing release v0.0.11:
...
+## Quality assurance
+
+We already have an extensive end-to-end test suite which covers most features
+and a good amount of edgecases, however it is impossible to eliminate every bug
+and regression this way because Upstream isn't self-contained and interactions
+with the OS and network.
+
+To be more confident that nothing has slipped thrugh our regular testing
+methods, and we don't ship a buggy release to our users we perform QA on the
+finished packages.
```
Comment (NunoAlexandre) [inline DEVELOPMENT.md:502]:
```suggestion
To be more confident that nothing has slipped through our regular testing
methods and we don't ship a buggy release to our users, we perform QA on the
finished packages.
```

**Review by NunoAlexandre — COMMENTED** [review]
This is great to have :1st_place_medal: 

It would be good to include in the QA a check against experimental features and the design system.

File: DEVELOPMENT.md [hunk 2]
```diff
@@ -482,6 +490,30 @@ Finalizing release v0.0.11:
 Release v0.0.11 successfully completed! 👏 🎉 🚀
 ```
 
+## Quality assurance
+
+We already have an extensive end-to-end test suite which covers most features
+and a good amount of edgecases, however it is impossible to eliminate every bug
+and regression this way because Upstream isn't self-contained and interactions
+with the OS and network.
```
[the remark this answers is not in the document]
Comment (rudolfs) [reply DEVELOPMENT.md:498]: Ya, still drafting...

[the remark this answers is not in the document]
Comment (NunoAlexandre) [reply DEVELOPMENT.md:498]: :construction:

**Comment by rudolfs** [comment]
> It would be good to include in the QA a check against experimental features and the design system.
> edit: which for me could close #910

@NunoAlexandre added a section about that: https://github.com/radicle-dev/radicle-upstream/pull/922/files#diff-d22b92cc1ad3b24930cf7de647c4433cR30-R32.

**Comment by rudolfs** [comment]
@xla found a way around https://github.com/radicle-dev/radicle-upstream/issues/932, maybe we don't need to change it for now.

**Review by sarahscott — APPROVED** [review]

File: QA.md [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+- [ ] download the binary package from the location provided above
+- [ ] remove any old configuration and user data
+  - on macOS: mount the `.dmg` and install Upstream in the default location
+    `/Applications`, then run:
+    ```
+    /Applications/radicle-upstream.app/Contents/MacOS/radicle-upstream --factory-reset
+    ```
+  - on Linux: run the `.AppImage` like so:
+    ```
+    /Applications/radicle-upstream.app/Contents/MacOS/radicle-upstream --factory-reset
```
Comment (xla) [inline QA.md:16]:
```suggestion
    /<PATH_TO_DOWNLOAD>/radicle-upstream-0.0.14.AppImage --factory-reset
```

**Review by xla — COMMENTED** [review]
Invaluable!

**Review by NunoAlexandre — APPROVED** [review]
:100:

File: QA.md [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+### Onboarding
+
+- [ ] Always shown when no identity is set up
+  - [ ] First app start
+  - [ ] Subsequent app starts if the identity creation was not completed
+  - [ ] Can't exit onboarding before identity is created and global keyboard
+        shortcuts are disabled
+- [ ] It's possible to go to the next screen by pressing <kbd>enter</kbd> if
+      the input field validations allow it
+- [ ] Handle and passphrase validations work
```
Comment (FintanH) [inline QA.md:51]: Might I suggest a link to validation examples, perhaps?

**Review by xla — APPROVED** [review]
🕑 🐌 👢 ➕

File: QA.md [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+    - [ ] Pushing new commits to Radicle via `git push rad` work (temporary
+          password until we have passphrases: `radicle-upstream`)
+    - [ ] Pulling changes work: make changes in the project folder you created
+          in project creation, push them to Radicle with `git push rad`, switch
+          to the checkout working directory, do a `git pull`
+
+
+#### Source browsing
+
+- [ ] Metadata and stats in UI reflect what is in the actual repository
```
Comment (FintanH) [inline QA.md:98]: Might be useful to have the `git` commands here to copy and paste to do the comparison

File: QA.md [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+- [ ] `README.md` files are shown by default and markdown is rendered as HTML
+  - [ ] Links to external resources open in external browser
+  - [ ] Links to internal resources don't do anything
+  - [ ] If the project doesn't have a readme, a placeholder is shown
```
Comment (FintanH) [inline QA.md:102]:
```suggestion
  - [ ] If the project doesn't have a `README.md`, a placeholder is shown
```

**Review by FintanH — COMMENTED** [review]
Looks good, left a few suggestions

File: QA.md (hunk 4 above)
Comment (rudolfs) [reply QA.md:51]: For posterity -- all @FintanH comments will be addressed in: https://github.com/radicle-dev/radicle-upstream/pull/941.

**Pull request merged**